cmd/soroban-rpc/internal/preflight/pool.go

```diff
@@ -24,21 +27,56 @@ type workerRequest struct {
 }
 
 type PreflightWorkerPool struct {
-	ledgerEntryReader db.LedgerEntryReader
-	networkPassphrase string
-	logger            *log.Entry
-	isClosed          atomic.Bool
-	requestChan       chan workerRequest
-	wg                sync.WaitGroup
+	ledgerEntryReader        db.LedgerEntryReader
+	networkPassphrase        string
+	logger                   *log.Entry
+	isClosed                 atomic.Bool
+	requestChan              chan workerRequest
+	concurrentRequestsMetric prometheus.Gauge
+	errorFullCounter         prometheus.Counter
+	durationMetric           *prometheus.SummaryVec
+	wg                       sync.WaitGroup
 }
 
-func NewPreflightWorkerPool(workerCount uint, jobQueueCapacity uint, ledgerEntryReader db.LedgerEntryReader, networkPassphrase string, logger *log.Entry) *PreflightWorkerPool {
+func NewPreflightWorkerPool(daemon interfaces.Daemon, workerCount uint, jobQueueCapacity uint, ledgerEntryReader db.LedgerEntryReader, networkPassphrase string, logger *log.Entry) *PreflightWorkerPool {
 	preflightWP := PreflightWorkerPool{
 		ledgerEntryReader: ledgerEntryReader,
 		networkPassphrase: networkPassphrase,
 		logger:            logger,
 		requestChan:       make(chan workerRequest, jobQueueCapacity),
 	}
+	requestQueueMetric := prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Namespace: daemon.MetricsNamespace(),
+		Subsystem: "preflight_pool",
+		Name:      "queue_length",
+		Help:      "number of preflight requests in the queue",
+	}, func() float64 {
+		return float64(len(preflightWP.requestChan))
+	})
+	preflightWP.concurrentRequestsMetric = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: daemon.MetricsNamespace(),
+		Subsystem: "preflight_pool",
+		Name:      "concurrent_requests",
+		Help:      "number of preflight requests currently running",
+	})
+	preflightWP.errorFullCounter = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: daemon.MetricsNamespace(),
+		Subsystem: "preflight_pool",
+		Name:      "queue_full_errors",
+		Help:      "number of preflight full queue errors",
+	})
+	preflightWP.durationMetric = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace: daemon.MetricsNamespace(),
+		Subsystem: "preflight_pool",
+		Name:      "request_duration",
```

Comment:
I think that you want to call it "request_ledger_get_duration_seconds" for consistency with the rest of the duration-like metrics. This isn't just the request duration, but rather measuring the total of the database operations.